apps/web/pages/api/cron/syncAppMeta.ts

```diff
@@ -0,0 +1,58 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getAppWithMetadata } from "@calcom/app-store/_appRegistry";
+import { prisma } from "@calcom/prisma";
+import type { AppCategories, Prisma } from "@calcom/prisma/client";
+
+/**
+ * syncAppMeta makes sure any app metadata that has been replicated into the database
+ * remains synchronised with any changes made to the app config files.
+ */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const apiKey = req.headers.authorization || req.query.apiKey;
+  if (process.env.CRON_API_KEY !== apiKey) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Invalid method" });
+    return;
+  }
+
+  console.log(`🧐 Checking DB apps are in-sync with app metadata`);
+
+  const dbApps = await prisma.app.findMany();
+
+  for await (const dbApp of dbApps) {
+    const app = await getAppWithMetadata(dbApp);
+    const updates: Prisma.AppUpdateManyMutationInput = {};
+
+    if (!app) {
+      console.error(`💀 App ${dbApp.slug} (${dbApp.dirName}) no longer exists.`);
+      continue;
+    }
+
+    if (
+      dbApp.categories.length !== app.categories.length ||
+      !dbApp.categories.every((category) => app.categories.includes(category))
+    ) {
+      updates["categories"] = app.categories as AppCategories[];
+    }
+
+    if (dbApp.dirName !== app.dirName ?? app.slug) {
+      updates["dirName"] = app.dirName ?? app.slug;
+    }
+
+    if (Object.keys(updates).length > 0) {
+      console.log(`🔨 Updating app ${dbApp.slug} with ${Object.keys(updates).join(", ")}`);
+      await prisma.app.update({
+        where: { slug: dbApp.slug },
+        data: updates,
+      });
```

Comment:
```suggestion
     /* await prisma.app.update({
        where: { slug: dbApp.slug },
        data: updates,
      });*/
```

Because we are doing a one time sync for the first time, there can be unknown issues. I would prefer it to be just a dry run initially and once we review that it's fine, we can do an actual run.

Reply:
Gotcha. Do you think feature flags would be better here? Makes it easier to toggle on / off without having to make code changes.

Would also provide a very quick & convenient way of disabling the sync if it ever does introduce problems.

Reply:
Feature flag is more of User facing and this is a dev requirement. This can be an env variable in my opinion, that way it won't affect self-hosters. We are going to verify the query and they might need to do that too if they want to setup that sync.